x/oracle/keeper.go

```diff
@@ -0,0 +1,357 @@
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.GetFeederDelegationKey(operator))
+	if bz == nil {
+		// by default the right is delegated to the validator itself
+		return sdk.AccAddress(operator)
+	}
+
+	return sdk.AccAddress(bz)
+}
+
+// SetFeederDelegation sets the account address that the validator operator delegated oracle vote rights to.
+func (k Keeper) SetFeederDelegation(ctx sdk.Context, operator sdk.ValAddress, delegatedFeeder sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.GetFeederDelegationKey(operator), delegatedFeeder.Bytes())
+}
+
+// IterateFeederDelegations iterates over the feed delegates and performs a callback function.
+func (k Keeper) IterateFeederDelegations(ctx sdk.Context,
+	handler func(delegator sdk.ValAddress, delegate sdk.AccAddress) (stop bool)) {
+
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.FeederDelegationKey)
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		delegator := sdk.ValAddress(iter.Key()[2:])
+		delegate := sdk.AccAddress(iter.Value())
+
+		if handler(delegator, delegate) {
+			break
+		}
+	}
+}
+
+// GetMissCounter retrieves the # of vote periods missed in this oracle slash window
+func (k Keeper) GetMissCounter(ctx sdk.Context, operator sdk.ValAddress) uint64 {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.GetMissCounterKey(operator))
+	if bz == nil {
+		// By default the counter is zero
+		return 0
+	}
+
+	var missCounter gogotypes.UInt64Value
+	k.cdc.MustUnmarshal(bz, &missCounter)
+	return missCounter.Value
+}
+
+// SetMissCounter updates the # of vote periods missed in this oracle slash window
+func (k Keeper) SetMissCounter(ctx sdk.Context, operator sdk.ValAddress, missCounter uint64) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&gogotypes.UInt64Value{Value: missCounter})
+	store.Set(types.GetMissCounterKey(operator), bz)
+}
+
+// DeleteMissCounter removes miss counter for the validator
+func (k Keeper) DeleteMissCounter(ctx sdk.Context, operator sdk.ValAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.GetMissCounterKey(operator))
+}
+
+// IterateMissCounters iterates over the miss counters and performs a callback function.
+func (k Keeper) IterateMissCounters(ctx sdk.Context,
+	handler func(operator sdk.ValAddress, missCounter uint64) (stop bool)) {
+
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.MissCounterKey)
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		operator := sdk.ValAddress(iter.Key()[2:])
+
+		var missCounter gogotypes.UInt64Value
+		k.cdc.MustUnmarshal(iter.Value(), &missCounter)
+
+		if handler(operator, missCounter.Value) {
+			break
+		}
+	}
+}
+
+// GetAggregateExchangeRatePrevote retrieves an oracle prevote from the store
+func (k Keeper) GetAggregateExchangeRatePrevote(ctx sdk.Context,
+	voter sdk.ValAddress) (aggregatePrevote types.AggregateExchangeRatePrevote,
+	err error) {
```

Comment:
formatting is off here.

Reply:
Going to remove the named return values, will comb through for all examples of that as well.